src/Layers/xrRender/FSkinned.cpp

```diff
@@ -765,27 +578,28 @@ void CSkeletonX_ext::_CollectBoneFaces(Fvisual* V, u32 iBase, u32 iCount)
         else
             R_ASSERT2(0, "not implemented yet");
     }
-
-#ifdef USE_DX9 // Don't use hardware buffers in DX10 since we can't read them
     break;
     case RM_SINGLE:
     case RM_SKINNING_1B:
     {
-        vertHW_1W* vertices = nullptr;
-        R_CHK(V->p_rm_Vertices->Lock(V->vBase, V->vCount, (void**)&vertices, D3DLOCK_READONLY));
+        VERIFY(V->vStride == sizeof(vertHW_1W));
+        void* data = static_cast<BYTE*>(V->p_rm_Vertices->Map(V->vBase, V->vCount * V->vStride, true));
+        vertHW_1W* vertices = static_cast<vertHW_1W*>(data);
+
         for (u32 idx = 0; idx < iCount; idx++)
         {
             vertHW_1W& v = vertices[indices[idx]];
             CBoneData& BD = Parent->LL_GetData(v.get_bone());
             BD.AppendFace(ChildIDX, (u16)(idx / 3));
         }
-        V->p_rm_Vertices->Unlock();
+        V->p_rm_Vertices->Unmap(); // ? keep ?
```

Comment:
To keep or not to keep?

Let me guess: the code is working good, but this is a potential place for optimization?

Reply:
Yes, you're right. :)
The idea behind this is simple. While loading animated meshes some kinematics setups access their geometry data. It's reasonable to free host heap buffers right after the meshes get processed but I not sure yet where is the best place for it.

Reply:
Aha... Ok then. We should note that we can improve this, but I believe, it's better to do this optimization in another PR)

Maybe, @SkyLoaderr can give us a piece of advice on this?